core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -424,14 +412,21 @@ private object RTS {
                       if (finalizer eq null) {
                         // No finalizer, so immediately produce the error.
                         curIo = null
-                        result = ExitResult.Failed(error)
+
+                        val ts: List[Throwable] = status.get match {
```

Comment:
We could clean this up by adding a method to status called `causes`, which extracts out `List[Throwable]` from the interrupt causes.